### DISCUSSION TITLE
Accessing the parent model

### DISCUSSION THREAD

**Discussion by anishsarkar** [filamentphp/filament#1155] [Help] [answered]
How does one access the parent model from a relationship form? 

I have some user groups, like faculty, student etc. In profile relationship of the user, if the user type is faculty, there are couple of extra fields, such as office, office_telephone etc,. So, I have made a profile form with those fields and want to show them conditionally if the user is of faculty type. 

Any pointers? Thanks in advance for help.

**Comment by danharrin** [comment, marked as answer]
Try this on the field/s you want to show:

```
->visible(fn ($livewire) => $livewire->ownerRecord->type === 'faculty')
```

https://filamentadmin.com/docs/2.x/forms/advanced#using-closure-customisation

**Reply by anishsarkar** [reply]
Thanks. It worked great! 

Can you also let me know in rules method what are the accessible variables? If I want to write some validation in the relationship, which might depend on the parent (ownerRecord), how do I access that?

**Reply by danharrin** [reply]
All variables are accessible everywhere. Check out the docs page I linked :)

**Reply by anishsarkar** [reply]
Thanks. I could get it working. I was trying to define a custom rule and pass the livewire or ownerRecord in the constructor of the rule, something like

> new MyCustomRule ($livewire) 

or some version of it. I now understand that I have to write a closure in which I can pass the livewire object and do the rest of thing as given in your example and I am able to get it working like that. Only thing is that I would have liked that this logic to be taken out of the Resource Object and be tucked away in the CustomRule object.

**Reply by anishsarkar** [reply]
Silly of me, should have realized that I could always do

> fn(Livewire $livewire) => new MyCustomRule ($livewire)

That does it nicely.
